internal/integration_test/spectest/spectest.go

```diff
@@ -0,0 +1,516 @@
+package spectest
+
```

Comment:
extracted the spec tests code into the public package so that they can be used for bulk memory spectests.